src/api.js

```diff
@@ -163,6 +199,20 @@ export default class Api {
       return Promise.resolve(results);
     return this.fetchServerSettings()
       .then(serverSettings => {
+        const maxRequests = serverSettings["cliquet.batch_max_requests"];
+        if (records.length > maxRequests) {
+          return Promise.all(partition(records, maxRequests).map(chunk => {
+            return this.batch(bucketName, collName, chunk, options);
+          }))
+            .then(batchResults => {
+              return batchResults.reduce((acc, batchResult) => {
```

Comment:
This would deserve a small comment IMO (e.g. assemble responses of chunked batch into one single result array)